internal/nginx-meshctl/deploy/deploy_test.go

```diff
@@ -43,7 +44,41 @@ var _ = Describe("Deploy", func() {
 		// we need to set either tracing/telemetry to nil
 		values.Telemetry = nil
 		values.Tracing = nil
-		Expect(deployer.Deploy()).To(Succeed())
+		_, err := deployer.Deploy()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("deploys air-gapped mesh", func() {
+		// mute stdout for this test since it will print the entire manifest
+		stdout := os.Stdout
+		defer func() { os.Stdout = stdout }()
+		os.Stdout = os.NewFile(0, os.DevNull)
+
+		var values helm.Values
+		Expect(yaml.Unmarshal(valuesYaml, &values)).To(Succeed())
+		deployer.Values = &values
+
+		// default deploy values are not valid
+		// we need to set tracing and telemetry to nil
+		values.Telemetry = nil
+		values.Tracing = nil
+
+		// Set up an air-gapped deployment.
+		values.Registry.Server = "test-registry"
+		values.Registry.DisablePublicImages = true
+		manifest, err := deployer.Deploy()
+		Expect(err).ToNot(HaveOccurred())
+
+		valid := isImageSourceValid(manifest, values.Registry.Server)
+		Expect(valid).To(BeTrue())
+
+		// Set up a non air-gapped deployment.
+		values.Registry.DisablePublicImages = false
+		manifest, err = deployer.Deploy()
+		Expect(err).ToNot(HaveOccurred())
+
+		valid = isImageSourceValid(manifest, values.Registry.Server)
+		Expect(valid).To(BeFalse())
```

Comment:
Shouldn't this be part of the `It("deploys without error"...` test on L#34 instead of combining 2 tests in 1?

Reply:
Oh, I was too slow and it got merged before I finished writing my comment. 😄